spds/constraints.md

```diff
@@ -4,4 +4,72 @@ title: Constraints
+Once the policy is enacted once and an adjustment is made there is no further adjustment for the next 4 time units.
+![example-cooldown-1.png](..%2Fimages%2Fconstraints%2Fexample-cooldown-1.png)
+
+In case the max number of scaling operations is set to 1, after the first adjustment, there is one additional operation allowed within the cooldown period but no more than one.
+![example-cooldown-2.png](..%2Fimages%2Fconstraints%2Fexample-cooldown-2.png)
+
+Cooldown constraints are applicable to the policy and help to allow the system to stabilize after another adjustment is made so that the effects are better observable.
+
+### Interval Constraint
+An interval constraint is defined by a time offset and a time period in which the policy is switched to an inactive state.
+For the same example, the picture below defines a time offset of 2 time units and a time period of 5 time units.
+During the offset period, the policy is active and adjustments are made normally. 
+After offset ends, the inactivity period starts that lasts for the specified _intervalDuration_.
+
+After the inactivity period ends and the simulation continues, the policy is switched to an active state again and adjustments are made normally.
+However, if the _repeat_ flag is set to true, the interval constraint is applied again with the _offset_ and _intervalDuration_ (shown with a dashed line in the figure).
+
+![example-interval.png](..%2Fimages%2Fconstraints%2Fexample-interval.png)
+
+### Thrashing Constraint
+
+Another temporal constraints that is applicable to the whole [TargetGroup](../docu/#targetsTargetGroup) and restricts the behavior of all the policies applying to it. 
+The main objective behind the thrashing constraint is to avoid oscillations in the system. 
+The thrashing constraint is defined by a minimum time period between two consecutive adjustments with opposite directions.
+The motivation behind defining a thrashing constraint can be the stability of the system and the billing type of the cloud provider.
+Often when resources are provisioned it does not make any sense to deprovision them after a short period of time while you are billed for the next 10 minutes. 
+Hence, the modeller can specify a Thrashing constraint.
+
+The picture below shows the behavior of a policy without a thrashing constraint in which a consecutive scale in is occurring after a scale out.
+![constraint-thrashing.png](..%2Fimages%2Fconstraints%2Fconstraint-thrashing.png)
+
+When adding a Thrashing constraint, the modeler can avoid such an oscillation. The picture below shows the behavior of the same policy with a thrashing constraint of 5 time units.
+![constraint-thrashing-2.png](..%2Fimages%2Fconstraints%2Fconstraint-thrashing-2.png)
+
+
+## State Based Constraints
+
+State based constraints are constraints in which the restriction and behavior alternation considers the model state during the simulation.
```

Comment:
see comment above.

Reply:
?

Reply:
the fuck, i wrote that comment, but it's not there o_O
iirc, i went something like this : https://www.oxfordlearnersdictionaries.com/definition/english/alternation?q=alternation  vs https://www.oxfordlearnersdictionaries.com/definition/english/alteration?q=alteration i think you actually mean alteration?